packages/firebase_data_connect/firebase_data_connect/lib/src/network/grpc_transport.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of firebase_data_connect_grpc;
+
+/// Transport used for Android/iOS. Uses a GRPC transport instead of REST.
+class GRPCTransport implements DataConnectTransport {
+  /// GRPCTransport creates a new channel
+  GRPCTransport(this.transportOptions, this.options, this.auth, this.appCheck) {
+    bool isSecure =
+        transportOptions.isSecure == null || transportOptions.isSecure == true;
+    channel = ClientChannel(transportOptions.host,
+        port: transportOptions.port ?? 443,
+        options: ChannelOptions(
+            credentials: (isSecure
+                ? const ChannelCredentials.secure()
+                : const ChannelCredentials.insecure())));
+    stub = ConnectorServiceClient(channel);
+    name =
+        'projects/${options.projectId}/locations/${options.location}/services/${options.serviceId}/connectors/${options.connector}';
+  }
+
+  /// FirebaseAuth
+  FirebaseAuth? auth;
+
+  /// FirebaseAppCheck
+  FirebaseAppCheck? appCheck;
+
+  /// Name of the endpoint.
+  late String name;
+
+  /// ConnectorServiceClient used to execute the query/mutation.
+  late ConnectorServiceClient stub;
+
+  /// ClientChannel used to configure connection to the GRPC server.
+  late ClientChannel channel;
+
+  /// Current host configuration.
+  @override
+  TransportOptions transportOptions;
+
+  /// Data Connect backend configuration options.
+  @override
+  DataConnectOptions options;
+
+  Future<Map<String, String>> getMetadata() async {
+    String? authToken;
+    try {
+      authToken = await auth?.currentUser?.getIdToken();
+    } catch (e) {
+      print('Unable to get auth token: ' + e.toString());
+    }
+    String? appCheckToken;
+    try {
+      authToken = await appCheck?.getToken();
+    } catch (e) {
+      print('Unable to get app check token: ' + e.toString());
+    }
+    Map<String, String> metadata = {
+      'x-goog-request-params': 'location=${options.location}&frontend=data',
+      'x-goog-api-client': 'gl-dart/flutter fire/$packageVersion'
+    };
+
+    if (authToken != null) {
+      metadata['x-firebase-auth-token'] = authToken;
+    }
+    if (appCheckToken != null) {
+      metadata['X-Firebase-AppCheck'] = appCheckToken;
+    }
```

Comment:
for authToken and appCheckToken
The operand can't be null, so the condition is always 'false'. Try removing the condition, an enclosing
          condition, or the whole conditional statement. • unnecessary_null_comparison